cli/azd/pkg/project/service_manager.go

```diff
@@ -328,6 +294,49 @@ func (sm *serviceManager) Package(
 			Service: serviceConfig,
 		}
 
+		hasBuildOutput := buildOutput != nil
+		restoreResult := &ServiceRestoreResult{}
+
+		// Get the language / framework requirements
+		frameworkRequirements := frameworkService.Requirements()
+
+		// When a previous restore result was not provided, and we require it
+		// Then we need to restore the dependencies
+		if frameworkRequirements.Package.RequireRestore && (!hasBuildOutput || buildOutput.Restore == nil) {
```

Comment:
This seems odd. Should we not be looking at the `restore` result? This seems to suggest that we would not respect the restore caching?

Reply:
We are pulling the restore result out of the build result to check if we previously restored.

Reply:
What if I restored without building? Or maybe that's not possible?